src/machine.rs

```diff
@@ -249,6 +249,37 @@ impl<'m> Machine<'m> {
         Ok(())
     }
 
+    /// Delete the machine.
+    ///
+    /// Deletes the machine, cleaning up all associated resources.
+    #[instrument(skip_all)]
+    pub async fn delete(self) -> Result<(), Error> {
+        let vm_id = self.config.vm_id();
+
+        if self.is_running() {
+            // Trying to shut down gracefully
+            self.shutdown().await?;
+            info!("{vm_id}: Waiting for the jailer to shut down...");
```

Comment:
"jailer" -> "VM process". We can't assume jailer is involved, even though currently that's the case.